File: 20210308/FirstForm/app/src/main/java/br/com/fiap/firstform/MainActivity.java
package br.com.fiap.firstform;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText edtName;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edtName = findViewById(R.id.edtName);
    }

    public void send(View view) {
        String name = edtName.getText().toString();
        
        if (name.trim().isEmpty()){
            Toast.makeText(this, "type the name correctly", Toast.LENGTH_SHORT).show();
        }
        else {
            Toast.makeText(this, "Hello " + name, Toast.LENGTH_SHORT).show();
        }
    }
}

File: 20210322/Desafio05Palindromos/app/src/main/java/br/com/fiap/desafio05palindromos/MainActivity.kt
package br.com.fiap.desafio05palindromos

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.EditText
import android.widget.Toast
import java.text.Normalizer

class MainActivity : AppCompatActivity() {

    lateinit var etInput: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        etInput = findViewById(R.id.etInput)

    }

    /**
     *  Evento disparado pelo botão verificarString
     *
     *  @param view
     */
    fun verificarString(view: View) {
        var strInput = etInput.text.toString().trim().toLowerCase()

        if (strInput.isEmpty()) {
            toast(getString(R.string.Informe_corretamente))
            return
        }

        val regex = "[^a-z]".toRegex()
        strInput = regex.replace(removeAcento(strInput), "")
        val strInvertida = strInput.reversed()

        val msg = if (strInput.equals(strInvertida)) getString(R.string.eh_palindormo) else getString(R.string.nao_eh_palindromo)

        toast(msg)
    }

    /**
     *  Função que substitui vogais com acentos verificarString
     *
     *  @param str
     *  @return String
     */
    fun removeAcento(str: String): String {
        var strSemAcento: String = Normalizer.normalize(str, Normalizer.Form.NFD)
        strSemAcento = strSemAcento.replace("[^\\p{ASCII}]", "")
        return strSemAcento
    }

    fun toast(msg: String) {
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()
    }

}